### PR TITLE
Add method to isposdef! for Hermitian and Symmetric

### LINKED ISSUE #422: isposdef not defined for symmetric matrices
The standard linalg function `isposdef` does not work for symmetric matrices (recent git master):

```
x = rand(5, 5)
x = x*x'
isposdef(x)
isposdef(Symmetric(x))
```

The last line fails with an error:

```
ERROR: MethodError: no method matching isposdef!(::Symmetric{Float64,Array{Float64,2}})
```

which I guess is caused by a missing definition. One workaround is to access underlying matrix directly but it is inconvenient:

```
isposdef(Symmetric(x).data)
```

I think this behavior is not correct and surprising for each user, I have not found any mention of such limitation in the documentation. There is an [additional implementation for Diagonal matrices](https://github.com/JuliaLang/julia/blob/db9709c6f282877f5afe3fd1fef45e0a459b81b0/base/linalg/diagonal.jl#L108) and this works fine:

```
isposdef(Diagonal(rand(1, 5)))
```

### LINKED ISSUE #21559: isposdef not defined for symmetric matrices
The standard linalg function `isposdef` does not work for symmetric matrices (recent git master):

```
x = rand(5, 5)
x = x*x'
isposdef(x)
isposdef(Symmetric(x))
```

The last line fails with an error:

```
ERROR: MethodError: no method matching isposdef!(::Symmetric{Float64,Array{Float64,2}})
```

which I guess is caused by a missing definition. One workaround is to access underlying matrix directly but it is inconvenient:

```
isposdef(Symmetric(x).data)
```

I think this behavior is not correct and surprising for each user, I have not found any mention of such limitation in the documentation. There is an [additional implementation for Diagonal matrices](https://github.com/JuliaLang/julia/blob/db9709c6f282877f5afe3fd1fef45e0a459b81b0/base/linalg/diagonal.jl#L108) and this works fine:

```
isposdef(Diagonal(rand(1, 5)))
```

### PR DESCRIPTION
**Pull request by fredrikekre**
Fix JuliaLang/LinearAlgebra.jl#422 

Perhaps too late for 0.6, but here it is

### CODE REVIEW

**Comment by KristofferC** [comment]
Instead of A.data shouldn't it be something like `triu(A.data)` (or `tril` depending on the `uplo` char). So "garbage" on the unused side gets ignored?

**Comment by andreasnoack** [comment]
@KristofferC You probably concluded that already but `xpotrf` ignores the other triange (and imaginary parts in the diagonal).

**Review by andreasnoack — APPROVED** [review]

File: test/linalg/symmetric.jl [hunk 1]
```diff
@@ -129,6 +129,12 @@ let n=10
         @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
         @test det(a + a.') ≈ det(Symmetric(a + a.', :L))
 
+        # isposdef
+        if eltya != BigFloat # isposdef not defined for Matrix{BigFloat}
+            @test isposdef(Symmetric(asym)) == isposdef(full(Symmetric(asym)))
```
Comment (ararslan) [inline test/linalg/symmetric.jl]: Might be good to directly test the methods for the mutating version, `isposdef!`, that you've added here.

Comment (fredrikekre) [inline test/linalg/symmetric.jl]: Guess we should test both.

**Review by ararslan — APPROVED** [review]

**Comment by fredrikekre** [comment]
Forgot to exclude `Int` for the in place testing, now it should work.

File: base/linalg/symmetric.jl [hunk 2]
```diff
@@ -264,6 +264,8 @@ det(A::Symmetric) = det(bkfact(A))
 inv{T<:BlasFloat,S<:StridedMatrix}(A::Hermitian{T,S}) = Hermitian{T,S}(inv(bkfact(A)), A.uplo)
 inv{T<:BlasFloat,S<:StridedMatrix}(A::Symmetric{T,S}) = Symmetric{T,S}(inv(bkfact(A)), A.uplo)
 
+isposdef!(A::HermOrSym{<:BlasFloat, <:StridedMatrix}) = ishermitian(A) && LAPACK.potrf!(A.uplo, A.data)[2] == 0
```
Comment (Sacha0) [inline base/linalg/symmetric.jl]: Extraneous space prior to `<:StridedMatrix`?

**Pull request merged**